### PR TITLE
Improve at-sync performance.

### PR DESCRIPTION
**Pull request by maleadt**
From https://discourse.julialang.org/t/most-efficient-way-of-waiting-for-gpu-results/19193
bors try

### CODE REVIEW

**Comment by maleadt** [comment]
bors try

**Comment by maleadt** [comment]
Hmm, not sure why the master version of CUDAdrv isn't used.
bors try

**Comment by maleadt** [comment]
bors try

**Comment by maleadt** [comment]
Finally.
bors r+

File: docs/make.jl [hunk 1]
```diff
@@ -1,5 +1,10 @@
 using Documenter, CuArrays
 
+using Pkg
+if haskey(ENV, "GITLAB_CI")
+    Pkg.develop.(["CUDAapi", "GPUArrays", "CUDAnative", "NNlib", "CUDAdrv"])
```
Comment (KristofferC) [inline docs/make.jl]: I would add these at the master branch instead (and also fuse it to a single call to `Pkg.add`)

```jl
Pkg.add([PackageSpec(name = x; rev = "master") for x in ["CUDAapi", "GPUArrays", "CUDAnative", "NNlib", "CUDAdrv"]])```

File: test/runtests.jl [hunk 2]
```diff
@@ -1,5 +1,12 @@
 using Test
 
+# development often happens in lockstep with other packages,
+# so check-out the master branch of those packages.
+using Pkg
+if haskey(ENV, "GITLAB_CI")
```
Comment (KristofferC) [inline test/runtests.jl]: Same here.

File: docs/make.jl (hunk 1 above)
Comment (maleadt) [reply docs/make.jl]: Ah yeah that's probably better, I had changed it to `develop` in an attempt to test the issue (where I did the `Pkg.add` as part of `Pkg.build`).

**Pull request merged**